--- configure.in.xbase	2010-11-14 05:23:47.000000000 +0000
+++ configure.in	2010-11-14 05:24:24.000000000 +0000
@@ -859,7 +859,7 @@
 	xbasedir=""
 	for d in $dir /usr /usr/local /opt/xbase
 	do
-		if test -f $d/include/xbase/xbase.h
+		if test -f $d/include/xbase64/xbase64.h
 		then
 			AC_MSG_RESULT(found Xbase in $d)
 			xbasedir=$d
@@ -871,7 +871,7 @@
 		AC_MSG_WARN(Xbase backend not used)
 	else
 		XBASE_CFLAGS=-I${xbasedir}/include
-		XBASE_LIBS="-L${xbasedir}/lib -lxbase"
+		XBASE_LIBS="-L${xbasedir}/lib -lxbase64"
 		AC_DEFINE(HAVE_XBASE, 1, [Have Xbase])
 	fi
 fi
--- providers/xbase/gda-xbase-database.cpp.xbase	2010-11-14 05:32:17.000000000 +0000
+++ providers/xbase/gda-xbase-database.cpp	2010-11-14 05:32:29.000000000 +0000
@@ -19,7 +19,7 @@
  * Software Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
  */
 
-#include <xbase/xbase.h>
+#include <xbase64/xbase64.h>
 #include "gda-xbase-database.h"
 #include "gda-xbase-provider.h"
 
